### ISSUE TITLE
Unable to re-launch NF with same instance ID 

### ISSUE THREAD

**Issue by catherinemeadows** [sdnfv/openNetVM#233]
## Bug Report

**Current Behavior**
When re-launching an NF with the same instance ID (from either command line or from a JSON config file), the manager exits with the errors: 
```
RING: Cannot reserve memory 
RING: Cannot reserve memory
RING: Cannot reserve memory
EAL: Error - exiting with code: 1
Cause: Cannot create rx ring queue for NF 1
```

and the NF hangs on 
```
APP: Waiting for manager to assign an ID...
```

This bug occurs when re-launching an NF with the same instance ID (even if the service ID is changed). 

**Expected behavior/code**
NFs should successfully re-launch if assigned the same instance ID as a prior launch. 

**Steps to reproduce**
Launch NF with specified instance ID (either command line or from a JSON config file), terminal with Ctrl+C, and re-launch with the same instance ID. 

Command line instructions for running speed_tester:

```
$ ./go.sh -- -n 1 -r 1 -- -d 2
```
Terminate with Ctrl + C and relaunch:
```
$ ./go.sh -- -n 1 -r 1 -- -d 2
```
<img width="1440" alt="Screen Shot 2020-06-16 at 11 29 22 AM" src="https://user-images.githubusercontent.com/43352540/84794759-a383dc00-afc4-11ea-84f2-b52adfa1bf58.png">

The same error occurs when re-launching an NF with the same instance ID but a different service ID: 
```
$ ./go.sh -- -n 1 -r 1 -- -d 2
```
Terminate with Ctrl + C and relaunch:
```
$ ./go.sh -- -n 1 -r 2 -- -d 2
```

**Environment**

- OS: Ubuntu 18.04
- onvm version:  20.05 

**Possible Solution**

Make sure rings are being freed on NF shutdown

**Additional context/Screenshots**

The screenshot shows the result of trying to re-launch speed_tester from a JSON config file with the same instance ID as the first successful launch: 
```
$ ./go.sh -F ../example_config.json -- -- -d 1
$ Ctrl + C
$ ./go.sh -F ../example_config.json -- -- -d 1
```

<img width="1440" alt="rx_ring_bug" src="https://user-images.githubusercontent.com/43352540/84792774-4a1aad80-afc2-11ea-974c-3f2bfef84ea0.png">

**Comment by catherinemeadows** [comment]
@bdevierno1 this might be a fix that can be included in your [PR](https://github.com/sdnfv/openNetVM/pull/216)

**Comment by dennisafa** [comment]
Yea this ones on me - when I moved the NF ring allocation from the manager to the NF, I didn't account for the fact thats rings need to be allocated from a secondary process. In my latest PR (NF pool) i wrote a function that allocates a single ring from the manager using a NF -> manager communication mechanism. When we merge that we should fix this issue by having the NF use that function whenever allocating the ring.

**Comment by kevindweb** [comment]
Fixed from merge in #272
